test/util/e2e.go

```diff
@@ -112,6 +115,39 @@ func CreateVisibilityClient(user string) visibilityv1beta1.VisibilityV1beta1Inte
 	return visibilityClient
 }
 
+func getRsAvailableConditionFrom(d *appsv1.Deployment) *appsv1.DeploymentCondition {
+	for i := range d.Status.Conditions {
+		c := d.Status.Conditions[i]
+		if c.Type == appsv1.DeploymentProgressing && c.Reason == "NewReplicaSetAvailable" {
+			return &c
+		}
+	}
+	return nil
+}
+
+func rolloutOperatorDeployment(ctx context.Context, k8sClient client.Client, key types.NamespacedName) {
+	deployment := &appsv1.Deployment{}
+
+	gomega.Expect(k8sClient.Get(ctx, key, deployment)).To(gomega.Succeed())
+	rsAvailableCondition := getRsAvailableConditionFrom(deployment)
+	gomega.Expect(rsAvailableCondition).ToNot(gomega.BeNil())
+	beforeUpdateTime := rsAvailableCondition.LastUpdateTime
+
+	gomega.EventuallyWithOffset(2, func(g gomega.Gomega) {
+		g.Expect(k8sClient.Get(ctx, key, deployment)).To(gomega.Succeed())
+		deployment.Spec.Template.ObjectMeta.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().Format(time.RFC3339)
+		g.Expect(k8sClient.Update(ctx, deployment)).To(gomega.Succeed())
+	}, Timeout, Interval).Should(gomega.Succeed())
+
+	gomega.EventuallyWithOffset(2, func(g gomega.Gomega) {
+		g.Expect(k8sClient.Get(ctx, key, deployment)).To(gomega.Succeed())
+		rsAvailableCondition := getRsAvailableConditionFrom(deployment)
+		g.Expect(rsAvailableCondition).ToNot(gomega.BeNil())
```

Comment:
I would suggest to implement this using `apimeta.FindStatusCondition`. When it fails it should produce a better diff, eg. tell us the actual "reason" field.

Reply:
Ah, as pointed out by @kaisoz offline we cannot use FindStatusCondition as the returned types don't match.

In that case I would prefer to have a helper function FindDeploymentCondition which only checks for the type (not reason), and then assert for the reason outside of the function. This way it will be more re-usable, and also we should have better error messages.